nixops/resources/gce_route.py

```diff
@@ -0,0 +1,181 @@
+        network_differ = route.network.split("/")[-1] != self.network
+
+        return real_state_differ or network_differ
+
+    def _get_machine_property(self, machine_name, property):
+        """Get a property from the machine """
+        machine = self.depl.get_machine(machine_name)
+
+        return getattr(machine, property)
+
+    def _check(self):
+
+        if self._route_is_missing():
+            self.state = self.MISSING
+            return False
+
+        if self._real_state_differ():
+                if self.depl.logger.confirm("Route properties are different from those in the state, "
+                                            "destroy route {0}?".format(self.route_name)):
+                    self._destroy_route()
+                    self.state = self.MISSING
+        return True
+
+    def create(self, defn, check, allow_reboot, allow_recreate):
+        self.copy_credentials(defn)
+
+        if check:
+            if self._route_is_missing():
+                self.state = self.MISSING
+
+            elif self._real_state_differ():
+                if allow_recreate:
+                    self._destroy_route()
+                    self.state = self.MISSING
+                else:
+                    self.warn("Route properties are different from those in the state,"
+                              " use --allow-recreate to delete the route and deploy it again.")
+
+        if defn.destination.startswith("res-"):
+            # if a machine resource was used for the destination, get
+            # the public IP of the instance into the definition of the
+            # route
+            machine_name = defn.destination[4:]
+            defn.destination = "{ip}/32".format(ip=self._get_machine_property(machine_name, "public_ipv4"))
+
+        if self.is_deployed() and self.properties_changed(defn):
+            if allow_recreate:
+                self.log("deleting route {0}...".format(self.route_name))
+                self._destroy_route()
+                self.state = self.MISSING
+            else:
+                raise Exception("GCE routes are immutable, you need to use --allow-recreate.")
+
+        if self.state != self.UP:
+            with self.depl._db:
+                self.log("creating {0}...".format(self.full_name))
+                self.copy_properties(defn)
+
+                if defn.nextHop.startswith("res-"):
+                    try:
+                        nextHop_name = self._get_machine_property(defn.nextHop[4:], "machine_name")
+                        defn.nextHop = self.connect().ex_get_node(nextHop_name)
+                    except AttributeError:
+                        raise Exception("nextHop can only be a GCE machine.")
+                        raise
+                    except libcloud.common.google.ResourceNotFoundError:
+                        raise Exception("The machine {0} isn't deployed, it need to be before it's added as nextHop".format(nextHop_name))
```

Comment:
Can we use create_after method instead ?

Reply:
I added create_after, but I'm keeping this in case we use --include the flag to only deploy the route. In that case the deployment the above exception will be raised.